Add comprehensive overview section to README (#4)
anthonychu/20250605-durabletask-aca-jobs: README.md

# Durable Task Scheduler with Azure Container Apps Jobs

## Overview

> The Durable Task Scheduler (DTS) is a distributed task scheduling framework that enables developers to write reliable, stateful workflows using familiar programming constructs. Azure Container Apps Jobs are serverless, containerized compute resources that run to completion and automatically scale based on demand.

Reviewer: ```suggestion
The Durable Task Scheduler is a distributed task scheduling framework that enables developers to write reliable, stateful workflows using familiar programming constructs. Azure Container Apps jobs are serverless, containerized compute resources, including GPUs, that run to completion and automatically scale based on demand.
```

> While Azure Container Apps Jobs can be triggered in various ways (manually, on schedule, or via events), orchestrating complex workflows—such as running jobs sequentially, implementing fan-out/fan-in patterns, or coordinating multiple interdependent tasks—presents significant challenges without proper orchestration tools.

Reviewer: ```suggestion
While Azure Container Apps jobs can be triggered in various ways (manually, on schedule, or via events), orchestrating complex workflows—such as running jobs sequentially, implementing fan-out/fan-in patterns, or coordinating multiple interdependent tasks—presents significant challenges without proper orchestration tools.
```

> This is where the Durable Task Scheduler excels: it provides a robust framework for orchestrating Container Apps Jobs with built-in support for workflow patterns like fan-out/fan-in, sequential execution, error handling, retries, and long-running stateful processes. By combining DTS with Azure Container Apps Jobs, you can build scalable, resilient workflows that automatically handle failures and provide visibility into execution progress.

Reviewer: ```suggestion
This is where the Durable Task Scheduler excels: it provides a robust framework for orchestrating Container Apps jobs with built-in support for workflow patterns like fan-out/fan-in, sequential execution, error handling, retries, and long-running stateful processes. By combining Durable Task Scheduler with Azure Container Apps jobs, you can build scalable, resilient workflows that automatically handle failures and provide visibility into execution progress.
```
